sw/device/lib/crypto/drivers/hmac.c

```diff
@@ -0,0 +1,84 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/drivers/hmac.h"
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/mmio.h"
```

Comment:
I think we should use `abs_mmio` for crypto drivers as well. @moidx WDYT?

Reply:
Yes, I think this is a good idea. I think it is also ok to update in a follow up PR, after instrumenting unittest support for `abs_mmio`.

Reply:
@jadephilipoom would you like to switch to `abs_mmio` in this PR or in a follow-up PR?

Reply:
I think it makes sense to update this PR, actually -- this PR is "copy the code from silicon_creator and then change it to not use abs_mmio", so reverting it back to abs_mmio will reduce code churn. I'll update this PR soon.

Reply:
Discussed this with Alphan offline; this is a slightly bigger job than it initially appears because `abs_mmio` and its unit-testing code will both need to be relocated, and the fact that the cryptolib is using mask ROM drivers still is blocking a change needed for mask ROM hardening (see https://github.com/lowRISC/opentitan/pull/10190#issuecomment-1017618142)

I think now it makes sense to merge this as-is, and I'll create an issue to switch it (and the OTBN driver that will follow) to `abs_mmio` later. That'll unblock me to move the cryptolib code out of `silicon_creator` so it doesn't interfere with the mask ROM changes.